tests/test_views.py

```diff
@@ -1772,3 +1772,111 @@ def test_get_course_enrollment_page_with_unparseable_course_effort(
+        enrollment_client.get_course_enrollment.return_value = None
+
+        # Get landing page
+        course_id = self.demo_course_id
+        enterprise_customer = EnterpriseCustomerFactory(
+            name='Starfleet Academy',
+            enable_data_sharing_consent=True,
+            enforce_data_sharing_consent='at_enrollment',
+        )
+        enterprise_landing_page_url = reverse(
+            'enterprise_course_enrollment_page',
+            args=[enterprise_customer.uuid, course_id],
+        )
+        self._login()
+        response = self.client.get(enterprise_landing_page_url)
+        assert response.status_code == 200
+
+        # Set up expected context
+        course_modes = [
+            {
+                'mode': 'professional',
+                'title': 'Professional Track',
+                'original_price': '$100',
+                'final_price': '$100',
+                'description': 'Earn a verified certificate!',
+                'premium': True,
+            }
+        ]
+        expected_context = {
+            'platform_name': 'edX',
+            'page_title': 'Choose Your Track',
+            'course_id': course_id,
+            'course_name': self.dummy_demo_course_details_data['name'],
+            'course_organization': self.dummy_demo_course_details_data['org'],
+            'course_short_description': self.dummy_demo_course_details_data['short_description'],
+            'course_pacing': 'Instructor-Paced',
+            'course_start_date': parse(self.dummy_demo_course_details_data['start']).strftime('%B %d, %Y'),
+            'course_image_uri': self.dummy_demo_course_details_data['media']['course_image']['uri'],
+            'enterprise_customer': enterprise_customer,
+            'enterprise_welcome_text': (
+                "<strong>Starfleet Academy</strong> has partnered with <strong>edX</strong> to "
+                "offer you high-quality learning opportunities from the world's best universities."
+            ),
+            'confirmation_text': 'Confirm your course',
+            'starts_at_text': 'Starts',
+            'view_course_details_text': 'View Course Details',
+            'select_mode_text': 'Please select one:',
+            'price_text': 'Price',
+            'continue_link_text': 'Continue',
+            'course_modes': course_modes,
+            'course_effort': '',
+            'level_text': 'Level',
+            'effort_text': 'Effort',
+            'duration_text': 'Average Length',
+            'course_overview': None,
+            'organization_logo': None,
+            'organization_name': None,
+            'duration': '12 weeks per course',
+            'course_level_type': 'Type 1',
+            'close_modal_button_text': 'Close',
+        }
+
+        # Compare expected context with response result
+        for key, value in expected_context.items():
+            assert response.context[key] == value  # pylint: disable=no-member
```

Comment:
@replaceafill There's quite a bit of duplication between these tests (and in this file in general). Just wanted to point this out -- since we need to move forward with this, I'm fine with deferring further work on refactoring the tests to another PR/ticket.